fe/fe-core/src/main/java/com/starrocks/catalog/ScalarType.java

```diff
@@ -65,6 +63,7 @@ public class ScalarType extends Type implements Cloneable {
     @SerializedName(value = "len")
     private int len = -1;
     private boolean isAssignedStrLenInColDefinition = false;
+    private boolean isSubquery = false;
```

Comment:
I think scalar type needn't care the isSubquery info

Reply:
> I think scalar type needn't care the isSubquery info

I know your design and means. maybe you need explain your whole plan and next job

I think members don't understand some design, because the process is different from other's expectations 

Process about this pr:
1. Expr trans to ScalarOperator phase, only wrap SubqueryExpr as an operator simply, none transfrom, so you don't konw the type of subquery, and all optimizations happened.
2. Parse subquery from ScalarOperator, and add ApplyOperator on logical plan
3. Rewrite expressions, replace all SubqueryOperator by the step 2 result

But the expect from other peoples, the process maybe:
1. Expr trans to ScalarOperator phase, and transform SubqueryExpr to ApplyNode, and save it in SubqueryOperator, then you knows the type of SubqueryOperator, then all optimizations happened.
2. Extract ApplyNode from SubqueryOperator, and add ApplyOperator on logical plan, and rewrite expressions by the result columns which is from ApplyNode